src/test/java/org/apache/maven/buildcache/xml/CacheConfigImplTest.java

```diff
@@ -122,11 +126,30 @@ private static void deepMockConfigFile(File mockFile, boolean exists) throws IOE
         when(mockPath.getFileSystem()).thenReturn(mockFileSystem);
         FileSystemProvider mockProvider = mock(FileSystemProvider.class);
         when(mockFileSystem.provider()).thenReturn(mockProvider);
+
+        // Mock for java < 20.
         if (!exists) {
             doThrow(new IOException("mock IOException"))
                     .when(mockProvider)
                     .checkAccess(ArgumentMatchers.eq(mockPath), ArgumentMatchers.any(AccessMode.class));
         }
+
+        // Mock for java >= 20. Since the FileSystemProvider.exists method does not exist before v20, we use reflection
+        // to create the mock
+        Optional<Method> methodToMock = Arrays.stream(FileSystemProvider.class.getDeclaredMethods())
+                .filter(method -> "exists".equals(method.getName()))
+                .findAny();
+        if (methodToMock.isPresent()) {
+            Class<?>[] paramTypes = methodToMock.get().getParameterTypes();
+            Object[] params = Arrays.stream(paramTypes)
+                    .map(paramType -> Mockito.any(paramType))
+                    .toArray();
+            try {
+                Mockito.when(methodToMock.get().invoke(mockProvider, params)).thenReturn(exists);
```

Comment:
Looks correct, but why need to Mock it at all? If it is not present in java 17, we are not using it in code. Is it used by JDK 20 internally?

Reply:
Yes, it is used internally by the static `java.nio.file.Files#exists` method.

Implementation differs from jdk versions.